src/pillowtalk_impl.c

```diff
@@ -630,13 +656,26 @@ static pt_response_t* http_operation(const char* http_method,
 
   curl_easy_setopt(curl_handle, CURLOPT_CONNECTTIMEOUT, 10);
 
+  /* specify header: Content-Type */
+  headers = curl_slist_append(headers, "Content-Type: application/json");
+  curl_easy_setopt(curl_handle, CURLOPT_HTTPHEADER, headers);
+
   // Want to avoid CURL SIGNALS
   curl_easy_setopt(curl_handle, CURLOPT_NOSIGNAL, 1);
 
   printf("%s : %s\n", http_method, server_target);
 
+  /* set the http verb */
   if (!strcmp("PUT", http_method))
+  {
     curl_easy_setopt(curl_handle, CURLOPT_UPLOAD, 1);
+    curl_slist_append(headers, "Transfer-Encoding: chunked");
+  }
+  else if (!strcmp("POST", http_method))
+  {
+    curl_easy_setopt(curl_handle, CURLOPT_POST, 1);
+    curl_slist_append(headers, "Transfer-Encoding: chunked");
+  }
```

Comment:
Oh, I just follow the example code from [here](https://curl.haxx.se/libcurl/c/post-callback.html).

It says that we have two way to send post data -- setting `chunked` header or setting `CURLOPT_POSTFIELDSIZE`.

I just pick up first method and it works for me, not sure which one it better.
(I do not test the second method).

I am willing to change the code if we figure out which one is better. :-)

```c
    /*
      If you use POST to a HTTP 1.1 server, you can send data without knowing
      the size before starting the POST if you use chunked encoding. You
      enable this by adding a header like "Transfer-Encoding: chunked" with
      CURLOPT_HTTPHEADER. With HTTP 1.0 or without chunked transfer, you must
      specify the size in the request.
    */ 
#ifdef USE_CHUNKED
    {
      struct curl_slist *chunk = NULL;
 
      chunk = curl_slist_append(chunk, "Transfer-Encoding: chunked");
      res = curl_easy_setopt(curl, CURLOPT_HTTPHEADER, chunk);
      /* use curl_slist_free_all() after the *perform() call to free this
         list again */ 
    }
#else
    /* Set the expected POST size. If you want to POST large amounts of data,
       consider CURLOPT_POSTFIELDSIZE_LARGE */ 
    curl_easy_setopt(curl, CURLOPT_POSTFIELDSIZE, pooh.sizeleft);
#endif
```